app/models/solr_document.rb

```diff
@@ -150,9 +150,37 @@ def extents
   end
 
   def fetch_html_safe(field)
+    return fetch_summary_note(field) if field == "summary_storage_note_ssm"
     fetch(field, []).map(&:html_safe)
   end
 
+  def fetch_summary_note(field)
+    note = fetch(field, [])
+    matches_abid_format = note.first.match(/^(?<location>.*: Boxes )(?:[A-Z]-\d{6}; )+/)
+    return note.map(&:html_safe) if note == [] || !matches_abid_format
+    boxes = note.first.scan(/[A-Z]-\d{6}/).sort
+    ["#{matches_abid_format[:location]}#{boxes_to_range(boxes)}"].map(&:html_safe)
+  end
+
+  def consecutive?(box1, box2)
+    same_prefix = box1.gsub(/\d/, "") == box2.gsub(/\d/, "")
+    return false unless same_prefix
+    box_numbers_consecutive = (box2.gsub(/\D/, "").to_i == box1.gsub(/\D/, "").to_i + 1)
+    box_numbers_consecutive
+  end
+
+  def consecutive_chunk(output, boxes)
+    first = boxes.first
+    boxes.shift while boxes.length > 1 && consecutive?(boxes[0], boxes[1])
```

Comment:
`boxes.present?` might be a bit more readable than `boxes.length > 1` since it doesn't have any spaces.